libuavcan/include/uavcan/time.hpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Types and utilities for working with time.
+ */
+/** @file */
+
+#ifndef UAVCAN_TIME_HPP_INCLUDED
+#define UAVCAN_TIME_HPP_INCLUDED
+
+#include "uavcan/uavcan.hpp"
+#include "uavcan/util/math.hpp"
+
+namespace uavcan
+{
+/**
+ * Protected base class for duration values.
+ *
+ * @tparam DURATIONTYPE  The datatype returned when retrieving durations from
+ *                       realizations of this base class. This type must be
+ *                       exactly 8 bytes in size and must support certain concepts
+ *                       implied by DurationBase.
+ */
+template <typename DURATIONTYPE>
+class DurationBase
+{
+    int64_t usec_; /**< Internal storage of the duration value in microseconds. */
```

Comment:
We should switch to `std::int*` instead of `uavcan::int` seeing as we can rely on the standard library now.

Reply:
(I'm confused) int64_t is from the standard library?

Reply:
This confuses me. int64_t is from the standard library. Do you mean I should use std::? That's probably correct.

Reply:
Yes, we must use `std::int64_t` because the C++ standard does not seem to define `::int64_t`.

Reply:
Okay. This is already changed in the latest version of the PR.

Reply:
This reminds me of one spectacular performance issue that I uncovered in PX4 some years ago which was caused by the usage of C standard library functions instead of `std::`: https://github.com/PX4/Firmware/pull/6829#issuecomment-286920967

Reply:
Interestingly enough, this is why we have to build gtest using `--gnu++11`